### PR TITLE
PR for frontend

### PR DESCRIPTION
**Pull request by CodeNamedRobin**

### CODE REVIEW

File: app/src/main/res/drawable/round_rect.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle" >
+
+    <!-- fill/color -->
+    <solid
+        android:color="#262626"/>
```
Comment (Pradyuman7) [inline app/src/main/res/drawable/round_rect.xml:7]: Is this going to be a color that we use at various places in our app? Then we should probably store it in colors.xml

File: app/src/main/res/values-land/dimens.xml
Comment (Pradyuman7) [inline app/src/main/res/values-land/dimens.xml:1]: Why do we have multiple values directories, and with their own dimens.xml?

File: app/src/main/res/layout/fragment_login.xml
Comment (TPGamesNL) [inline app/src/main/res/layout/fragment_login.xml:1]: image
image

like we saw during the meeting

Comment (TPGamesNL) [inline app/src/main/res/layout/fragment_login.xml:1]: When I rotate the screen, the EditTexts get cleared (e.g. I enter email, rotate screen, then email is gone)

Comment (TPGamesNL) [reply app/src/main/res/layout/fragment_login.xml:1]: It also goes to back to the login screen when you're on the register screen and rotate the phone

File: app/src/main/res/layout/fragment_login.xml [hunk 2]
```diff
@@ -1,56 +1,116 @@
...
\ No newline at end of file
+        android:background="@color/purple_500"
+        android:fillViewport="true"
+        tools:layout_editor_absoluteX="0dp"
+        tools:layout_editor_absoluteY="-294dp">
+
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent">
```
Comment (TPGamesNL) [inline app/src/main/res/layout/fragment_login.xml:20]: IDE gives a warning about this line (don't listen to its suggested fix, it gives infinite scroll)

File: app/src/main/res/layout/fragment_login.xml [hunk 3]
```diff
@@ -1,56 +1,116 @@
...
+                    app:layout_constraintTop_toBottomOf="@+id/register_button" />
+            </androidx.constraintlayout.widget.ConstraintLayout>
+
+            <TextView
+                android:id="@+id/textView2"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="40dp"
+                android:fontFamily="@font/kenia_regular"
+                android:text="Music \nMap"
```
Comment (TPGamesNL) [inline app/src/main/res/layout/fragment_login.xml:104]: Hardcoded string

File: app/src/main/res/layout/fragment_register.xml [hunk 4]
```diff
@@ -1,96 +1,192 @@
...
+
+            </androidx.constraintlayout.widget.ConstraintLayout>
+
+            <TextView
+                android:id="@+id/textView2"
+                android:layout_width="322dp"
+                android:layout_height="106dp"
+                android:layout_marginTop="40dp"
+                android:fontFamily="@font/kenia_regular"
+                android:text="MusicMap"
```
Comment (TPGamesNL) [inline app/src/main/res/layout/fragment_register.xml:180]: Hardcoded string

File: app/src/main/res/layout/fragment_login.xml [hunk 5]
```diff
@@ -1,56 +1,116 @@
...
+                android:textAlignment="center"
+                android:textColor="@color/white"
+                android:textSize="90dp"
```
Comment (TPGamesNL) [inline app/src/main/res/layout/fragment_login.xml:107]: Use sp for text size, not dp

File: app/build.gradle [hunk 6]
```diff
@@ -27,18 +27,23 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+    buildFeatures {
+        viewBinding true
```
Comment (TPGamesNL) [inline app/build.gradle:31]: What do we need this for?

File: app/src/main/res/layout/fragment_register.xml
Comment (TPGamesNL) [inline app/src/main/res/layout/fragment_register.xml:1]: image
these should be aligned

Comment (TPGamesNL) [reply app/src/main/res/layout/fragment_register.xml:1]: check with tablet:
image

Comment (TPGamesNL) [reply app/src/main/res/layout/fragment_register.xml:1]: Same for buttons on login fragment

File: app/src/main/res/values-w600dp/dimens.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <dimen name="fab_margin">48dp</dimen>
```
Comment (TPGamesNL) [inline app/src/main/res/values-w600dp/dimens.xml:2]: I don't see the `fab_margin` dimension used anywhere

File: app/src/main/res/layout/fragment_login.xml [hunk 8]
```diff
@@ -1,56 +1,116 @@
...
+        android:fillViewport="true"
+        tools:layout_editor_absoluteX="0dp"
+        tools:layout_editor_absoluteY="-294dp">
+
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent">
+
+            <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (TPGamesNL) [inline app/src/main/res/layout/fragment_login.xml:22]: Is it fine to just use a ConstraintLayout inside of a ConstraintLayout?

**Review by TPGamesNL — CHANGES_REQUESTED** [review]

**Comment by TPGamesNL** [comment]
Btw there are build errors with the fonts, because both master and your branch added the same font

File: app/src/main/res/drawable/round_rect.xml (hunk 1 above)
Comment (CodeNamedRobin) [reply app/src/main/res/drawable/round_rect.xml:7]: This is not a color that's going to be used a lot I believe, but I can add it anyways.

File: app/build.gradle (hunk 6 above)
Comment (CodeNamedRobin) [reply app/build.gradle:31]: No clue why that was added, removed.

File: app/src/main/java/com/example/musicmap/screens/auth/AuthActivity.java [hunk 9]
```diff
@@ -24,15 +25,19 @@ public class AuthActivity extends AppCompatActivity {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
+        System.out.println("Created activity.");
```
Comment (TPGamesNL) [inline app/src/main/java/com/example/musicmap/screens/auth/AuthActivity.java:28]: If this is a debug message for testing, remove it. If it should be kept, use Log.d/i/w/e instead

File: app/src/main/res/layout-land/fragment_register.xml
Comment (TPGamesNL) [inline app/src/main/res/layout-land/fragment_register.xml:1]: image
I think the back and register buttons should be on the same horizontal line

**Review by TPGamesNL — APPROVED** [review]

**Pull request merged**